packages/ts/react-signals/src/FullStackSignal.ts

```diff
@@ -0,0 +1,167 @@
+import { signal } from '@preact/signals-react';
+import type { ConnectClient, Subscription } from '@vaadin/hilla-frontend';
+import { nanoid } from 'nanoid';
+import { Signal } from './core.js';
+
+const ENDPOINT = 'SignalsHandler';
+
+type SubscriptionEvent<T> = Readonly<{
+  id: string;
+  type: StateEventType;
+  value: T;
+}>;
+
+/**
+ * Types of changes that can be produced or processed by a signal.
+ */
+export enum StateEventType {
+  SET = 'set',
+  SNAPSHOT = 'snapshot',
+}
+
+/**
+ * An object that describes the change of the signal state.
+ */
+export type StateEvent<T> = Readonly<{
+  id: string;
+  type: StateEventType;
+  value: T;
+}>;
+/**
+ * An object that describes a data object to connect to the signal provider
+ * service.
+ */
+export type ConnectionData = Readonly<{
+  /**
+   * The client instance to be used for communication.
+   */
+  client: ConnectClient;
+
+  /**
+   * The name of the signal provider service endpoint.
+   */
+  endpoint: string;
+
+  /**
+   * The name of the signal provider service method.
+   */
+  method: string;
+}>;
+
+class SubscriptionManager<T> {
+  readonly #id: string;
+  readonly #data: ConnectionData;
+  #subscription?: Subscription<StateEvent<T>>;
+
+  constructor(id: string, data: ConnectionData) {
+    this.#data = data;
+    this.#id = id;
+  }
+
+  get subscription() {
+    return this.#subscription;
+  }
+
+  subscribe() {
+    const { client, endpoint, method } = this.#data;
+
+    this.#subscription ??= client.subscribe(ENDPOINT, 'subscribe', {
+      providerEndpoint: endpoint,
+      providerMethod: method,
+      clientSignalId: this.#id,
+    });
+
+    return this.#subscription;
+  }
+
+  async call(event: SubscriptionEvent<T>): Promise<void> {
```

Comment:
I suggest renaming `call` to `update`: 
```suggestion
  async update(event: SubscriptionEvent<T>): Promise<void> {
```

Reply:
Also, the `SubscriptionEvent` should change to `StateEvent`, here we not actually subscribing anything, but sending `SET` events.